src/textual_fspicker/select_directory.py

```diff
@@ -13,57 +13,27 @@
 from textual import on
 from textual.app import ComposeResult
 from textual.reactive import var
-from textual.widgets import Button, Label
+from textual.widgets import Button, Label, Input
 
 ##############################################################################
 # Local imports.
 from .base_dialog import ButtonLabel, FileSystemPickerScreen
 from .parts import DirectoryNavigation
-
+from .path_maker import MakePath
 
 ##############################################################################
-class CurrentDirectory(Label):
-    """A widget to show the current directory.
-
-    This widget is used inside a
-    [`SelectDirectory`][textual_fspicker.SelectDirectory] dialog to display
-    the currently-selected directory.
-    """
+class SelectDirectory(FileSystemPickerScreen):
+    """A directory selection dialog."""
 
-    DEFAULT_CSS = """
-    CurrentDirectory {
-        width: 1fr;
-        height: 3;
-        border: tall $background;
-        padding-left: 1;
-        padding-right: 1;
+    DEFAULT_CSS = FileSystemPickerScreen.DEFAULT_CSS + """
+    SelectDirectory InputBar {
+        Input { /* Style for the new path input */
```

Comment:
I'd discourage comments that explain how code has changed; this is what commit histories are for.